### PR TITLE
[fuchsia] Use scenic allocator service

### PR DESCRIPTION
**Pull request by dreveman**
Switch to scenic allocator and Image3 API.

### CODE REVIEW

File: shell/platform/fuchsia/flutter/vulkan_surface_pool.cc [hunk 1]
```diff
@@ -41,6 +41,10 @@ VulkanSurfacePool::VulkanSurfacePool(vulkan::VulkanProvider& vulkan_provider,
   sysmem_allocator_->SetDebugClientInfo(GetCurrentProcessName(),
                                         GetCurrentProcessId());
   FML_DCHECK(status != ZX_OK);
+  status = fdio_service_connect(
```
Comment (arbreng) [inline shell/platform/fuchsia/flutter/vulkan_surface_pool.cc:44]: Ah, I need to plumb these through the ComponentContext....will do in another CL

**Review by arbreng — APPROVED** [review]

**Comment by dreveman** [comment]
> @arbreng The test failure looks relevant to the patch. Can we update this patch?

That failure is expected until flutter engine is using a version of Fuchsia that includes https://fuchsia.googlesource.com/fuchsia/+/d7e580eb8060fcdc3e6941298dc143518ddb6d74 for these tests. How can I tell what version of Fuchsia is used for these tests?
